core/src/main/java/org/opensearch/sql/analysis/TypeEnvironment.java

```diff
@@ -29,14 +29,30 @@ public class TypeEnvironment implements Environment<Symbol, ExprType> {
   private final TypeEnvironment parent;
   private final SymbolTable symbolTable;
 
+  @Getter
+  private final SymbolTable reservedSymbolTable;
+
+  /**
+   * Constructor with empty symbol tables.
+   *
+   * @param parent parent environment
+   */
   public TypeEnvironment(TypeEnvironment parent) {
     this.parent = parent;
     this.symbolTable = new SymbolTable();
+    this.reservedSymbolTable = new SymbolTable();
   }
 
+  /**
+   * Constructor with empty reserved symbol table.
+   *
+   * @param parent parent environment
+   * @param symbolTable type table
+   */
   public TypeEnvironment(TypeEnvironment parent, SymbolTable symbolTable) {
     this.parent = parent;
     this.symbolTable = symbolTable;
+    this.reservedSymbolTable = new SymbolTable();
```

Comment:
Why is a separate symbol table needed? Why can't we add metafields to `symbolTable`?

Reply:
To separate out 'reserved fields' (that are in all tables) from fields that are specific to the index.  The code needs to know the difference between these two lists later on.